File: i2c.py
I2C_ADDR = 2

import mraa
from time import sleep

i2c = mraa.I2c(0)
i2c.address(I2C_ADDR)

# read 6 bytes from the arduino
for i in range(20):
	message = i2c.read(5)
	print(message)
	sleep(1) # delay for 1 second

File: i2c.c
#include <stdio.h>
#include <unistd.h>
#include <mraa/i2c.h>

#define #I2C_ADDR 2

int main() {
	char message[6];
	int i;

	mraa_init();
	mraa_i2c_context i2c;

	i2c = mraa_i2c_init(0);
	mraa_i2c_address(i2c, I2C_ADDR);

	for (i = 0; i < 20; i++) {
		mraa_i2c_read(i2c, message, 6);
		printf("%.6s\n", message);
		sleep(1);
	}

	mraa_i2c_stop(i2c);
	return 0;
}